addons/web/static/src/scss/bootstrap_review_frontend.scss

```diff
@@ -15,6 +15,14 @@
     @include text-emphasis-variant(".text-#{$color}", $value);
 }
 
+// TODO: To be removed in master (18.1)
+// Define `text-primary-emphasis` class here in order to take priority when in
+// conjunction with other text-[x] classes
+@at-root {
```

Comment:
What's the point of `@at-root` here @stefanorigano ?

Reply:
@qsm-odoo , to scope the temporary `$-emphasis-primary` variable, so it will not be accessible elsewhere.
It's unlikely  that someone will use a custom variable with that exact name, but I think that's good practice.
Let me know if it's ok, otherwise I may propose something like:

```scss
$-emphasis-primary: if(has-enough-contrast($primary, $body-bg), $primary, increase-contrast($primary, $body-bg));
@include text-emphasis-variant(".text-primary-emphasis", $-emphasis-primary);
$-emphasis-primary: null; // Reset the temporary variable

Reply:
@stefanorigano Oh ok... I have no strong opinion against it but I definitely don't think it should be considered a good practice to use anything for what it is not made for. It's just confusing (proof: I was confused). I would even prefer `@if true {` instead of `@at-root {` in this case: it clearly shows that this has no other purpose than scoping. But it's ugly too.

I probably would not care about scoping in that case. That's actually why I start some variable name with `-`, meaning "if a dev also named one like that later in such a way it might use one that was defined previously... then he is badly coding anyway". But if you really want to scope, what I would recommend it to create a mixin and call it directly. It's the same pattern we have in JS, creating anonymous functions sometimes. But in this case there is better: just don't define the variable.

Instead of:
```scss
@if at-root {
    $-emphasis-primary: if(has-enough-contrast($primary, $body-bg), $primary, increase-contrast($primary, $body-bg));
    @include text-emphasis-variant(".text-primary-emphasis", $-emphasis-primary);
}
```
Do
```scss
@include text-emphasis-variant(
    ".text-primary-emphasis",
    if(has-enough-contrast($primary, $body-bg), $primary, increase-contrast($primary, $body-bg))
);
```
Exact same number of LOC, without the confusion.

In the end, I was about to merge anyway but I am now wondering why you are not using `primary-text-emphasis` here instead of recomputing the same thing (you force primary here instead of the bootstrap default but why not force it in the definition of `primary-text-emphasis` then? or why changing `primary-text-emphasis` at all if it is not to use it?).

Reply:
Thanks @qsm-odoo , recomputing is indeed not necessary, `$primary-text-emphasis` is accessible and correctly adapted at this stage 👍